### DISCUSSION TITLE
Are there any plans for duckdb_vss extension to be available to duckdb_wasm?

### DISCUSSION THREAD

**Discussion by raspi0124** [duckdb/duckdb-wasm#1896] [Q&A] [answered]
I'm currently implementing a vector search using DuckDB and integrating it to work in the browser. However, I've recently discovered that the `duckdb_vss` extension, which provides vector search functionality for DuckDB, is not available for the Wasm platform.

My questions are:

1. Are there any plans to make the `duckdb_vss` extension available for DuckDB-Wasm?
2. If not, are there any alternative ways to implement vector search functionality in DuckDB-Wasm?
3. Is there a timeline or roadmap for adding vector search capabilities to the WASM version of DuckDB?

Any information or guidance on this matter would be greatly appreciated. (Partial answers are also really welcomed)
Thank you!

**Comment by carlopi** [comment, marked as answer]
`vss` does work in DuckDB-Wasm!

It should be a matter of:
```
LOAD vss;
```
What error message are you getting?

Check: https://shell.duckdb.org/#queries=v0,LOAD-vss~,CREATE-TABLE-my_vector_table-(vec-FLOAT%5B3%5D)~%0AINSERT-INTO-my_vector_table-SELECT-array_value(a%2C-b%2C-c)-FROM-range(1%2C-10)-ra(a)%2C-range(1%2C-10)-rb(b)%2C-range(1%2C-10)-rc(c)~%0ACREATE-INDEX-my_hnsw_index-ON-my_vector_table-USING-HNSW-(vec)~%0A%0A,SELECT-*-FROM-my_vector_table-ORDER-BY-array_distance(vec%2C-%5B1%2C-2%2C-3%5D%3A%3AFLOAT%5B3%5D)-LIMIT-3~%0A%0A,EXPLAIN-SELECT-*-FROM-my_vector_table-ORDER-BY-array_distance(vec%2C-%5B1%2C-2%2C-3%5D%3A%3AFLOAT%5B3%5D)-LIMIT-3~%0A%0A
Where I execute a bunch of random statements from the docs: https://duckdb.org/docs/extensions/vss.html

**Comment by raspi0124** [comment]
Thank you for your helpful response! I've resolved the issue by updating DuckDB to the latest version, and now the vector search functionality is working perfectly.
Again, thank you very much for your detailed support, and sorry for the late reply.
